### ISSUE TITLE
WebUI randomly refreshes without being prompted to

### ISSUE THREAD

**Issue by Cre8or** [syncthing/syncthing#9561]
### What happened?

(**Edit 1**: apologies for the unfinished report; I accidentally hit enter while typing.)

This issue first occured around 1-2 months ago and is now affecting 3 out of my 4 devices (the 4th, unaffected device is an Android phone, running v1.27.6).

When interacting with Syncthing though the web UI, the page will just randomly reload - even when there are menus/dialogs open. This is very frustrating, as it will discard any uncommitted changes that I was working on, such as complex ignore patterns or file versioning rules.

As far as logs go, all 3 affected devices only list the mobile device's connection timing out. I attempted turning it off altogether (thus getting rid of the log messages), and while the log messages have cleared up, the issue still persists. There does not seem to be anything in the logs that correlates to the page reloading, in terms of timestamps. The issue seems to occur periodically, sometimes up to 3 times in a minute.

**Edit 2**: Another user reported experiencing this issue [over on Reddit](https://www.reddit.com/r/Syncthing/comments/1bkdstm/webui_constantly_refreshing/). Unfortunately it doesn't seem like it's been resolved. 

### Syncthing version

v1.27.3

### Platform & operating system

Arch Linux (x86-64) / OpenWRT (ARM-64)

### Browser version

Firefox 126

**Comment by bt90** [comment]
Anything logged in the browser console or happening request wise?

**Comment by Cre8or** [comment]
Not much; when the page refresh occurs, all I get is:

![image](https://github.com/syncthing/syncthing/assets/3143797/68e6056f-bd16-44e9-9d02-e55df6217c93)

For comparison, a normal (manual) page refresh looks like this:

![image](https://github.com/syncthing/syncthing/assets/3143797/589ea763-6c7f-4522-a477-6faf3cb68b6d)

**Comment by Cre8or** [comment]
On my third affected device (my OpenWRT router), since I not accessing the web UI through the machine itself, but rather over the network via my Linux system, the behaviour is different. Instead of reloading, the page shows a connection error (as if the daemon was shut down). The browser console then looks like this:

![image](https://github.com/syncthing/syncthing/assets/3143797/32b4e742-e095-47be-ba1d-9f9d2e2b8548)

The daemon however is not actually down, as I am able to manually reload the page to get the WebUI to work again. Not as bad as outright discarding any pending changes, but still annoying.

**Comment by Cre8or** [comment]
I just tested on a clean installation of ungoogled chromium. The issue still occurs on the local machine's WebUI, but doesn't seem to appear on the remote WebUI.

**Comment by AudriusButkevicius** [comment]
I don't think we do anything special depending on whether it's viewed locally or from a remote machine, so that still sounds like something specific to that machine.

Try running syncthing on the other machine and see if you can reproduce that locally.

**Comment by valaphee** [comment]
It's noteworthy that the event XHR request fails because of 403 Forbidden,

which causes the page to reload https://github.com/syncthing/syncthing/blob/94181ade230f252e2f7a51cb7f4fd0c757c086da/gui/default/syncthing/core/eventService.js#L44C1-L44C35

**Comment by bt90** [comment]
Hm. Maybe it's cookie related?

**Comment by Cre8or** [comment]
I initially thought it was, because uMatrix had disabled cookies for the WebUI. However the issue still happens:
- with cookies enabled on uMatrix
- with uMatrix turned off
- with the uMatrix addon completely disabled
- with the fresh install of ungoogled Chromium (which has no addons)

I might've missed a test case, but I suspect it's probably not cookie related.

Another finding from looking at the network graph of the remote WebUI is that the endpoint request eventually fails with "CSRF Error", triggering a page reload (likely via the file @valaphee posted).

**Comment by bt90** [comment]
That cookie warning looks suspicious.

**Comment by bt90** [comment]
This might be related to #7399. Maybe Chromium is a bit more strict?

**Comment by Cre8or** [comment]
Huh. In that case, maybe the CSRF Cookie expires after a while, which causes the backend to return 403 when Syncthing attempts to do a periodic update, and that in turn causes the page to reload.

In which case: is this an intended feature? And can it be turned off?

**Comment by kallama** [comment]
I had this exact same problem on a brand new docker install using linuxserver/syncthing:1.27.7.

I was able to solve it by opening 22000 tcp/udp and 21027 udp as I was blocking those ports on my host.

**Comment by bt90** [comment]
We're not setting the `SameSite` attribute for the `CSRF-TOKEN-XXXXXXX` and `sessindd-XXXXXX` cookies. Maybe Ungoogled Chromium is a bit more strict in that regard?

Not sure if this intended or not, but the CSRF cookie is also missing the `Secure` attribute.

![grafik](https://github.com/syncthing/syncthing/assets/1926175/61e2cdad-0404-498b-acca-df6cedd6152b)

**Comment by TupaNegreiros** [comment]
I have the same problem here, only with a headless Raspberry Pi OS ARM32 with version v1.27.12 (this refreshing is happening for months).
I've tried a few browsers in 2 computers. I also noticed some 403 HTTP errors in console just before refreshing page, and a lot of "CSRF-TOKEN-XXXXXXX" cookie (with the same name). On other servers, there is only one CSRF-TOKEN-XXXXXXX and one sessionid-XXXXXX.
